### PR TITLE
Fix loaded model not taken to cuda device

### LINKED ISSUE #2189: Model training should be in GPU but model parameters are in CPU
**Describe the bug**
I'm trying to train a model using `allennlp train` in GPU in v0.7.2 but it fails when it's about to start training with the following stack trace:

```
Traceback (most recent call last):
  File "env/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "env/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "env/lib/python3.6/site-packages/allennlp/run.py", line 18, in <module>
    main(prog="allennlp")
  File "env/lib/python3.6/site-packages/allennlp/commands/__init__.py", line 72, in main
    args.func(args)
  File "env/lib/python3.6/site-packages/allennlp/commands/train.py", line 111, in train_model_from_args
    args.force)
  File "env/lib/python3.6/site-packages/allennlp/commands/train.py", line 142, in train_model_from_file
    return train_model(params, serialization_dir, file_friendly_logging, recover, force)
  File "env/lib/python3.6/site-packages/allennlp/commands/train.py", line 351, in train_model
    metrics = trainer.train()
  File "env/lib/python3.6/site-packages/allennlp/training/trainer.py", line 755, in train
    train_metrics = self._train_epoch(epoch)
  File "env/lib/python3.6/site-packages/allennlp/training/trainer.py", line 525, in _train_epoch
    self.optimizer.step()
  File "env/lib/python3.6/site-packages/torch/optim/adagrad.py", line 92, in step
    state['sum'].addcmul_(1, grad, grad)
RuntimeError: Expected object of backend CPU but got backend CUDA for argument #4 'tensor1'
```

It works if I set the device to -1 (CPU) or if I use allennlp v0.7.1.

Debugging a bit, I found that this is the change that breaks my code:

https://github.com/allenai/allennlp/pull/1944/files/3bed856fafa07a2a47d512a6102dacc8b12cf856#r241959869

merged <2mo ago. That change seems a bug to me. Or am I missing something?

**To Reproduce**
`allennlp train` some model in GPU.

**Expected behavior**
It should train without crashing.

**System (please complete the following information):**
 - OS: Linux
 - Python version: 3.6.7 - miniconda
 - AllenNLP version: 0.7.2
 - PyTorch version: both 0.4.1 and 1.0.0

### PR DESCRIPTION
**Pull request by bryant1410**
Fix #2189

### CODE REVIEW

File: allennlp/training/trainer.py [hunk 1]
```diff
@@ -1048,6 +1048,9 @@ def from_params(cls,  # type: ignore
         grad_clipping = params.pop_float("grad_clipping", None)
         lr_scheduler_params = params.pop("learning_rate_scheduler", None)
 
+        if isinstance(cuda_device, int) and cuda_device >= 0:
+            model = model.cuda(cuda_device)
```
Comment (matt-gardner) [inline allennlp/training/trainer.py]: I'm not sure what the right thing to do is when you're using multiple GPUs - I'm not really familiar with how the multi-GPU code works.  Anyone have any idea?

**Comment by matt-gardner** [comment]
I'm going to merge this soon unless someone pipes up and says something should change to better handle the multi-GPU case.

**Comment by bryant1410** [comment]
What if you open a new issue for the multi-GPU case?

File: allennlp/training/trainer.py (hunk 1 above)
Comment (matt-peters) [reply allennlp/training/trainer.py]: In the multiple GPU case, move the model to `cuda_device[0]` as is currently done in `Trainer.__init__` now (https://github.com/allenai/allennlp/blob/master/allennlp/training/trainer.py#L322)

Comment (matt-gardner) [reply allennlp/training/trainer.py]: Thanks!

**Comment by matt-gardner** [comment]
Thanks @bryant1410!

**Pull request merged**